drv_cfg.go

```diff
@@ -54,25 +55,38 @@ type ioctlGUID struct {
 	netIDTime  uint32
 }
 
+// Syscaller is an interface for syscall.Syscall
+type Syscaller interface {
+	Syscall(trap, a1, a2, a3 uintptr) (uintptr, uintptr, syscall.Errno)
+}
+
 // DrvCfgIsSDCInstalled will check to see if the SDC kernel module is loaded
 func DrvCfgIsSDCInstalled() bool {
-	if SCINIMockMode == true {
+	if SCINIMockMode {
 		return true
 	}
 	// Check to see if the SDC device is available
-	info, err := os.Stat(SDCDevice)
+	info, err := statFileFunc(SDCDevice)
 	if err != nil {
 		return false
 	}
 	return !info.IsDir()
 }
 
+var statFileFunc = func(path string) (os.FileInfo, error) {
+	return os.Stat(path)
+}
+
+var openFileFunc = func(name string) (*os.File, error) {
+	return os.Open(filepath.Clean(name))
+}
+
 // DrvCfgQueryGUID will return the GUID of the locally installed SDC
-func DrvCfgQueryGUID() (string, error) {
-	if SCINIMockMode == true {
+func DrvCfgQueryGUID(syscaller Syscaller) (string, error) {
```

Comment:
This is not ideal to have the user of the API additionally create and pass a syscaller which we only need for unit-testing. It should be an internal package variable with a default value that is overridable by the unit-tests. Syscaller should not be part of the library API.

Reply:
Fixed. Moved syscaller interface out of the API